core/example/benchmark/Cabana_BinSortPerformance.cpp

```diff
@@ -0,0 +1,217 @@
+/****************************************************************************
+ * Copyright (c) 2018-2019 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "Cabana_BenchmarkTimer.hpp"
+
+#include <Cabana_Core.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <numeric>
+#include <random>
+#include <sstream>
+#include <string>
+#include <vector>
+
+//---------------------------------------------------------------------------//
+// Performance test.
+template <class Device>
+void performanceTest( std::ostream &stream, const std::string &test_prefix )
+{
+    // Declare problem sizes.
+    std::vector<int> problem_sizes = {1000, 10000, 100000, 1000000, 10000000};
+    int num_problem_size = problem_sizes.size();
+
+    // For each problem size generate a random set of keys.
+    std::vector<Kokkos::View<unsigned long *, Device>> keys( num_problem_size );
+    for ( int i = 0; i < num_problem_size; ++i )
+    {
+        Kokkos::View<unsigned long *, Kokkos::HostSpace> host_keys(
+            "host_keys", problem_sizes[i] );
+        std::minstd_rand0 generator( 3439203991 );
+        for ( int n = 0; n < problem_sizes[i]; ++n )
+            host_keys( n ) = generator();
+        keys[i] =
+            Kokkos::View<unsigned long *, Device>( "keys", problem_sizes[i] );
+        Kokkos::deep_copy( keys[i], host_keys );
+    }
```

Comment:
That's silly.  Just generate the largest size and use subviews for smaller problems.

Reply:
fixed